payment/delete-transaction.go

```diff
@@ -43,20 +53,28 @@ func DeleteTransactionFx(w http.ResponseWriter, r *http.Request) (string, interf
 
 	if !isRenew {
 		transaction = tr.GetTransactionByUid(uid, "")
+		if transaction == nil {
+			log.Printf("transaction '%s' not found", uid)
+			return "", nil, fmt.Errorf("transaction '%s' not found", uid)
+		}
+		policy, err = plc.GetPolicy(transaction.PolicyUid, "")
 	} else {
 		collection = lib.RenewTransactionCollection
 		transaction = trxRenew.GetRenewTransactionByUid(uid)
+		if transaction == nil {
+			log.Printf("transaction '%s' not found", uid)
+			return "", nil, fmt.Errorf("transaction '%s' not found", uid)
+		}
+		policy, err = renew.GetRenewPolicyByUid(transaction.PolicyUid)
 	}
-
-	if transaction == nil {
-		log.Printf("transaction '%s' not found", uid)
-		return "", nil, fmt.Errorf("transaction '%s' not found", uid)
+	if err != nil {
+		return "", nil, err
```

Comment:
here the error should always be a 'policy not found'. Shall we log it as in the other error handlings?